### PR TITLE
fix(node): adjust type of `output` in return of `spawnSync`

### LINKED ISSUE #42221: @types/node@12.12.25 incorrect type of SpawnSyncReturns<T>["output"]
- [x] I tried using the `@types/node` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] Authors: @andrewbranch @SimonSchick .

I used `spawnSync` function from nodejs `child_process` module.
It seems that the type of `SpawnSyncReturns<T>["output"]` is not correct.
It is defined as the following: 
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a68259ee8883cc9899463fe79a78bb49a9d2f6de/types/node/v12/child_process.d.ts#L428

But when I run `spawnSync("someBinary", ["--version"])`
I get this:
image

So this should better be `(null | Buffer)[]` or maybe it depends on `T`? I am not sure...

Reproduction:
```ts
require("child_process").spawnSync("ls", ["--version"]).output
```

### PR DESCRIPTION
**Pull request by G-Rath**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: The node docs for [`spawnSync`](https://nodejs.org/api/child_process.html#child_process_child_process_spawnsync_command_args_options) say that the type of `output` is "`<Array>`", *but* they also say that `stdout` and `stderr` will be the value of `output[1]` & `output[2]` respectively, and that the type of those two properties will be `string | Buffer`. 
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Have tested this all the way back to Node v6 to confirm, including that the `encoding` option affects the value - I suspect this could actually be a tuple of `[null | T, T, T]` but the docs don't indicate that strongly enough for me to make that change given that I can't really prove it.

Closes #42221

### CODE REVIEW

File: types/node/v15/child_process.d.ts [hunk 1]
```diff
@@ -473,7 +473,7 @@ declare module 'child_process' {
     }
     interface SpawnSyncReturns<T> {
         pid: number;
-        output: string[];
+        output: Array<T | null>;
```
Comment (SimonSchick) [inline types/node/v15/child_process.d.ts:476]: You can remove the v15 changes, I intent to remove the typings for v15 as it is not supported: https://nodejs.org/en/about/releases/

**Comment by SimonSchick** [comment]
I think it makes sense to make it a tuple, I looked around the code a little and the array seems to be a list of writable streams, it may be more if the process had additional output pipes but.. is that a thing?

**Comment by G-Rath** [comment]
> but.. is that a thing?

That's what I wasn't sure of :joy:

How about we land it as is, and I'll make a followup PR to switch it to a tuple so it can be reverted separately if it causes problems.

**Comment by SimonSchick** [comment]
```js
console.log(spawnSync('node', ['test.js'], {
    encoding: 'utf-8',
    input: 'test',
    stdio: ['pipe', 'pipe', 'pipe', 'pipe', 'pipe', 'pipe']
}));

Yields:
```js
{
  status: 0,
  signal: null,
  output: [ null, 'asd\n', 'asdf\nasdfg\n', '', '', '' ],
  pid: 17472,
  stdout: 'asd\n',
  stderr: 'asdf\nasdfg\n'
}
```

with my test script being:

```js
console.log('asd');
console.warn('asdf');
console.error('asdfg');

const rl = require('readline');

const int = new rl.Interface(process.stdin)
int.question('yes?', (data) => {
    console.log(data);
    process.exit(0);
})
```

Sooo. not a tuple...

Further fun:

```js
console.log(spawnSync('node', ['test.js'], {
    encoding: 'utf-8',
    input: 'test',
    stdio: ['inherit', 'inherit', 'inherit']
}));
```

yields:

```js
{
  status: 0,
  signal: null,
  output: [ null, null, null ],
  pid: 5720,
  stdout: null,
  stderr: null
}
```

The cp module has been incredibly hard to type right... I think your change is good.

**Review by SimonSchick — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by G-Rath** [comment]
Ready to merge

**Pull request merged**